frontend/lib/identification/user_code_store.dart

```diff
@@ -24,7 +28,19 @@ class UserCodeStore {
 
   Future<List<DynamicUserCode>> load() async {
     const storage = FlutterSecureStorage();
-    final String? userCodesBase64 = await storage.read(key: _userCodesBase64Key);
+    String? userCodesBase64;
+    // make fake read on android - workaround for https://github.com/mogol/flutter_secure_storage/issues/653
+    if (Platform.isAndroid) {
+      String? userCodesBase64FirstRead = await storage.read(key: _userCodesBase64Key);
+      userCodesBase64 = await storage.read(key: _userCodesBase64Key);
+      // report error if first read is null but second read contains data to collect sentry data for reproduction
+      if (userCodesBase64FirstRead == null && userCodesBase64 != null) {
+        reportError('First read from secure storage is null but user data exists', null);
+      }
+    } else {
+      userCodesBase64 = await storage.read(key: _userCodesBase64Key);
+    }
+
     if (userCodesBase64 == null) return [];
     return userCodesBase64
         .split(_storageDelimiter)
```

Comment:
I have a question regarding the `importLegacyCard` function below. 

It constructs `UserCodeModel` without initializing it and then directly runs `insertCode` which in turn runs `UserCodeStore.store()`.
This seems quite bad.

I think it is not the cause of the problem, as `insertCode` is only called when a legacy code is found and thus should only be called once ever from importLegacyCard. But we should definitely refactor this.
`importLegacyCard` should return `Future<DynamicUserCode?>` and there should be a separate function `deleteLegacyCard` which should delete the key. The `initialize`  function of `UserCodeModel` should take care of calling these functions.

Also, we should add assertions to the functions of UserCodeModel which ensure they can only be called if UserCodeModel is initialized.

Reply:
i'm gonna create a task for it, thx for the hint